pkg/ansible/watches/watches.go

```diff
@@ -104,65 +101,65 @@ func parseLabelSelector(dls tempLabelSelector) metav1.LabelSelector {
 }
 
 type tempRequirement struct {
-	Key      string                       `yaml:"key"`
-	Operator metav1.LabelSelectorOperator `yaml:"operator"`
-	Values   []string                     `yaml:"values,omitempty"`
+	Key      string                       `json:"key"`
+	Operator metav1.LabelSelectorOperator `json:"operator"`
+	Values   []string                     `json:"values,omitempty"`
 }
 
-// UnmarshalYAML - implements the yaml.Unmarshaler interface for Watch.
-// This makes it possible to verify, when loading, that the GroupVersionKind
-// specified for a given watch is valid as well as provide sensible defaults
-// for values that were omitted.
-func (w *Watch) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	// Use an alias struct to handle complex types
-
-	type alias struct {
-		Group                       string                    `yaml:"group"`
-		Version                     string                    `yaml:"version"`
-		Kind                        string                    `yaml:"kind"`
-		Playbook                    string                    `yaml:"playbook"`
-		Role                        string                    `yaml:"role"`
-		Vars                        map[string]interface{}    `yaml:"vars"`
-		MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
-		ReconcilePeriod             string                    `yaml:"reconcilePeriod"`
-		ManageStatus                bool                      `yaml:"manageStatus"`
-		WatchDependentResources     bool                      `yaml:"watchDependentResources"`
-		WatchClusterScopedResources bool                      `yaml:"watchClusterScopedResources"`
-		Blacklist                   []schema.GroupVersionKind `yaml:"blacklist"`
-		Finalizer                   *Finalizer                `yaml:"finalizer"`
-		Selector                    tempLabelSelector         `yaml:"selector"`
-	}
-	var tmp alias
+// Use an alias struct to handle complex types
+type alias struct {
+	Group                       string                    `yaml:"group"`
+	Version                     string                    `yaml:"version"`
+	Kind                        string                    `yaml:"kind"`
+	Playbook                    string                    `yaml:"playbook"`
+	Role                        string                    `yaml:"role"`
+	Vars                        map[string]interface{}    `yaml:"vars"`
+	MaxRunnerArtifacts          int                       `yaml:"maxRunnerArtifacts"`
+	ReconcilePeriod             *metav1.Duration          `yaml:"reconcilePeriod,omitempty"`
+	ManageStatus                *bool                     `yaml:"manageStatus,omitempty"`
+	WatchDependentResources     *bool                     `yaml:"watchDependentResources,omitempty"`
+	WatchClusterScopedResources *bool                     `yaml:"watchClusterScopedResources,omitempty"`
+	Blacklist                   []schema.GroupVersionKind `yaml:"blacklist,omitempty"`
```

Comment:
we have no tests to ensure that. WDYT about add a test to check if the Blacklist will have the GKV informed after the parse ?
```suggestion
	Blacklist                   []schema.GroupVersionKind `yaml:"blacklist,omitempty"`
```